src/main/java/net/atos/entng/calendar/controllers/TrustedUrlController.java

```diff
@@ -0,0 +1,141 @@
+package net.atos.entng.calendar.controllers;
+
+import fr.wseduc.rs.*;
+import fr.wseduc.security.ActionType;
+import fr.wseduc.security.SecuredAction;
+import fr.wseduc.webutils.request.RequestUtils;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.json.JsonObject;
+import net.atos.entng.calendar.core.constants.Actions;
+import net.atos.entng.calendar.core.constants.Field;
+import net.atos.entng.calendar.services.ServiceFactory;
+import net.atos.entng.calendar.services.TrustedUrlService;
+import org.entcore.common.http.filter.ResourceFilter;
+import org.entcore.common.http.filter.SuperAdminFilter;
+import org.entcore.common.http.filter.Trace;
+import org.entcore.common.mongodb.MongoDbControllerHelper;
+import org.vertx.java.core.http.RouteMatcher;
+
+import java.util.Map;
+
+public class TrustedUrlController extends MongoDbControllerHelper {
+
+    private final TrustedUrlService trustedUrlService;
+
+    @Override
+    public void init(Vertx vertx, JsonObject config, RouteMatcher rm,
+                     Map<String, fr.wseduc.webutils.security.SecuredAction> securedActions) {
+        super.init(vertx, config, rm, securedActions);
+    }
+    public TrustedUrlController(String collection, ServiceFactory serviceFactory) {
+        super(collection);
+        this.trustedUrlService = serviceFactory.urlService();
+    }
+
+    @Get("/trusted/:idUrl")
+    @ApiDoc("Retrieve a trusted url by id")
+    @SecuredAction(value = "", type = ActionType.RESOURCE)
+    @ResourceFilter(SuperAdminFilter.class)
+    public void retrieveTrustedUrl(HttpServerRequest request){
+        final String id = request.params().get(Field.IDURL);
+        trustedUrlService.retrieve(id)
+                .onSuccess(res -> {
+                    JsonObject object = new JsonObject()
+                            .put(Field.RESULT, res)
+                            .put(Field.STATUS, Field.OK);
+                    renderJson(request, object);
+                })
+                .onFailure(err -> {
+                    JsonObject object = new JsonObject()
+                            .put(Field.REASON, err.getMessage())
+                            .put(Field.STATUS, Field.KO);
+                    renderError(request, object);
```

Comment:
Ici tu donne l'information de l'erreur en retour. Cela peut remonter des infos à des mauvaises personnes qui auront juste a appeler l'API. Tu peux notée l'erreur dans le log si tu veux.